tests/cpp/topk_test_kernels.cu

```diff
@@ -0,0 +1,251 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+// Define nvfuser_index_t at global scope for runtime files
+using nvfuser_index_t = int64_t;
+
+#include <tests/cpp/topk_test_helper.h>
+
+// Need to be included before argsort because of the dependency
+// from argsort
+namespace nvf {
+#include <runtime/index_utils.cu>
+} // namespace nvf
+
+#include <runtime/topk.cu>
+
+// Standard C++ headers
+#include <cstdint>
+
+// CUDA headers
+#include <cuda_bf16.h>
+#include <cuda_runtime.h>
+
+// Test framework headers
+#include <gtest/gtest.h>
+
+namespace nvfuser {
+
+//============================================================================
+// Fixed template parameter kernels (avoiding dynamic template instantiation)
+//============================================================================
+
+// Basic topk test kernel with configurable block size
+template <int BLOCK_SIZE, typename DataT, int ITEMS_PER_THREAD>
+__global__ void basicTopkTestKernel(
+    DataT* input,
+    DataT* output_values,
+    int64_t* output_indices,
+    int k,
+    bool largest) {
+  DataT input_data[ITEMS_PER_THREAD];
+  DataT top_values[ITEMS_PER_THREAD];
+  int64_t top_indices[ITEMS_PER_THREAD];
+
+  int thread_id = threadIdx.x;
+  int global_offset = thread_id * ITEMS_PER_THREAD;
+
+  // Load input data for this thread
+  for (int i = 0; i < ITEMS_PER_THREAD; i++) {
+    input_data[i] = input[global_offset + i];
+  }
+
+  // Call blockTopk with configurable block size template parameter
+  nvf::topk::blockTopK<BLOCK_SIZE, 1, 1, 0, 0, 0, DataT, ITEMS_PER_THREAD>(
+      top_values, top_indices, input_data, k, largest, true, blockDim);
```

Comment:
i.e. the blockTopK handled IO on register, it shouldn't require a global buffer size of the input.

Reply:
Oh, I see what you mean. I forgot the test global function uses registers for calling `blockTopK`. Updated the code. Thanks!